### PR TITLE
feat(doctor): `director doctor` install self-check

### PR DESCRIPTION
**Pull request by colinsurprenant**
Adds `director doctor`, a read-only install self-check that answers the one question the fail-safe hooks can't: **will Director actually fire?** The shims exit 0 when they can't resolve the binary, so a broken install is invisible — `doctor` walks the same resolution ladder the shims walk and turns that silent dead state into a loud, checkable one.

## What it checks

- **binary** — the `DIRECTOR_BIN` → `director` on PATH → install-symlink ladder the shims walk, including a `DIRECTOR_BIN` pinned in settings.json's `env` block (what Claude Code injects into the hook)
- **claude code hooks** — Director's tagged entries present in settings.json and the shims materialized
- **codex hooks** — reported only when a Codex install is present
- **hub** — present and writable, or creatable (its nearest existing ancestor is writable)

Exits non-zero when the install is broken; warns without failing on a partial one (e.g. a terminal-only install the desktop app would miss). On native Windows it reports the CLI-only state and points at WSL, rather than emitting failures whose only remedy (`director install`) also refuses.

## Review

Ran the standing convention — codex adversarial pass + `ce:code-reviewer`:

- **codex** flagged the settings.json-`env` `DIRECTOR_BIN` blind spot (P1, a real false-healthy — fixed via `install.SettingsDirectorBin`), hub ancestor writability (P2 — fixed, matches the ancestor-walk in `736cda4`), plus resolution-message faithfulness (PATH-before-symlink order).
- **ce:code-reviewer** approved, and caught the native-Windows remediation loop (fixed) and an exit-code doc overstatement (fixed).
- Deferred as an open-item (codex P1-3): verifying all four managed hook entries are wired, not just that one tagged entry exists — the write side is atomic, so the real risk is low.

## Gate

`go test ./... -race` green; `gofmt`/`vet` clean. New tests cover the settings-pin false-healthy end to end, the native-Windows path, the hub ancestor walk, and `SettingsDirectorBin` edge cases.

Part of the Phase-0 seal-the-funnel diffusion arc (step 0.2).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/director/doctorcmd.go [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+			"director is not on your PATH and the install symlink %s is missing or broken — the hooks will silently no-op. Re-run `director install` (or put director on your PATH).", in.binPath)}
+	}
+}
+
+// claudeHooksCheck verifies the Claude Code side is wired: the tagged entries are
+// in settings.json AND the shims those entries point at actually exist.
+func claudeHooksCheck(in doctorInputs) check {
+	if !install.ManagedEntriesPresent(in.settingsPath) {
+		return check{"claude code hooks", levelFail, fmt.Sprintf("no Director hooks in %s — run `director install`.", in.settingsPath)}
+	}
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply cmd/director/doctorcmd.go:239]: Fixed in d61974c. `ManagedEntriesPresent` collapses a read/parse failure to the same `false` as "no managed entries", so a malformed settings.json got the misleading "run `director install`" remedy — which `install` itself refuses on a malformed file. Added `install.SettingsParseError` and split the message: an unreadable or malformed file now reports as such, with the parse detail, and points at fixing the file. Covered by `TestDoctorMalformedSettingsReported`.

File: cmd/director/doctorcmd.go [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+
+// binResolves mirrors the shims' `[ -x "$bin" ] || command -v "$bin"`: a value
+// with a path separator must be an executable file; a bare name must be on PATH.
+func binResolves(v string) bool {
+	if strings.ContainsRune(v, filepath.Separator) || strings.Contains(v, "/") {
+		return isExecutable(v)
+	}
+	_, err := exec.LookPath(v)
+	return err == nil
+}
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply cmd/director/doctorcmd.go]: The divergence is real, but it's the safe direction and not faithfully fixable — so I kept PATH-only and expanded the comment (d61974c) rather than replicating the `-x` branch.

The shims' `[ -x "$bin" ]` for a bare name resolves against the **hook's** working directory (Claude Code's — the project dir). `director doctor` runs from the user's shell cwd, a *different* directory it has no way to know. Replicating that check here would probe the wrong directory and could report **healthy when the hook is actually dead** (a false-healthy, the dangerous direction). The divergence it does produce is a false-*alarm* (doctor says broken, a cwd-relative shim might run it), which is the tolerable one.

A bare-name `DIRECTOR_BIN` that resolves only via cwd is also unsupportable for a hook (it breaks under every other cwd), and Go's `exec.LookPath` deliberately excludes cwd for exactly that footgun. PATH is the portable, cwd-independent contract, so doctor models that.

File: cmd/director/doctorcmd.go [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+		if _, err := os.Stat(p); err == nil {
+			return p
+		}
+		parent := filepath.Dir(p)
+		if parent == p {
+			return p // reached the root; nothing more to climb
+		}
+		p = parent
+	}
+}
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply cmd/director/doctorcmd.go:384]: Fixed in 0c757ab. Real and reachable — the concrete case is a dangling-symlink ancestor: `Stat` follows it to not-exist and the old walk climbed past it, but `MkdirAll` can't create through a dangling link, so the hub read as creatable while the first write would fail (a false-healthy).

Replaced `nearestExistingDir` with `nearestExistingAncestor`, which mirrors `event.logTrulyAbsent`'s not-exist-vs-broken discipline: a non-not-exist stat error (EACCES / ENOTDIR / ELOOP), a dangling symlink (caught by an `Lstat` tiebreak), or an existing non-directory is a hard boundary the hub check now fails on rather than climbing past. Covered by `TestNearestExistingAncestor` and `TestHubDanglingSymlinkAncestorFails`.

File: internal/install/install.go [hunk 4]
```diff
@@ -402,6 +403,56 @@ func managedEntriesPresent(path string) bool {
...
 
+// ExpectedShims returns the basenames of the hook shim scripts install writes
+// into the hooks dir. The embedded shims/ dir is the single source of truth, so a
+// consumer that verifies an install (`director doctor`) checks exactly the set
+// install materializes — no drift between what is written and what is checked.
+func ExpectedShims() []string {
+	entries, err := fs.ReadDir(shimFS, "shims")
+	if err != nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply internal/install/install.go:414]: The nil-on-error branch is unreachable in a real binary: `shimFS` is an `embed.FS` compiled into the binary, and `//go:embed shims/*.sh` fails the *build* if it matches no files, so `fs.ReadDir(shimFS, "shims")` can't fail at runtime and the set is guaranteed non-empty.

Rather than add runtime loud-failure plumbing for a state that can't occur, I locked the invariant at the layer where it can actually break — the embed — with `TestExpectedShims` (0c757ab): it asserts the exact expected shim set, so any future change that empties or alters the embedded shims fails loudly in CI.

File: cmd/director/doctorcmd.go [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+	default:
+		return check{"binary", levelFail, fmt.Sprintf(
+			"director is not on your PATH and the install symlink %s is missing or broken — the hooks will silently no-op. Re-run `director install` (or put director on your PATH).", in.binPath)}
+	}
+}
+
+// claudeHooksCheck verifies the Claude Code side is wired: the tagged entries are
+// in settings.json AND the shims those entries point at actually exist.
+func claudeHooksCheck(in doctorInputs) check {
+	if !install.ManagedEntriesPresent(in.settingsPath) {
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply cmd/director/doctorcmd.go:229]: Tracked and deliberately deferred — this is the same gap codex raised (its P1-3). The maintainer reviewed it and chose to defer it as open-item `01KWYXM7ZE`, for this reason: `director install` writes all four managed entries atomically (all-or-nothing), so a partially-wired settings.json requires a hand-edit, which makes the false-healthy unlikely in practice.

The planned fix, when it lands, is to have `claudeHooksCheck` enumerate install's `directorEntries` (event + matcher + shim command path) and verify each is present, reusing install as the single source of truth rather than a bare "some tagged entry exists" probe. Left as-is for this PR.

File: cmd/director/doctorcmd.go [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+		// ManagedEntriesPresent collapses a read/parse failure to the same false as
+		// "no managed entries", but the remedies differ: a malformed settings.json
+		// needs fixing (install would refuse to overwrite it), not a re-run. Split
+		// the two so the message points at the real fix.
+		if err := install.SettingsParseError(in.settingsPath); err != nil {
+			return check{"claude code hooks", levelFail, fmt.Sprintf(
+				"%s is unreadable or not valid JSON (%v) — fix the file, then re-run `director install`.", in.settingsPath, err)}
+		}
+		return check{"claude code hooks", levelFail, fmt.Sprintf("no Director hooks in %s — run `director install`.", in.settingsPath)}
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply cmd/director/doctorcmd.go:238]: Confirmed and tracked as a follow-up. You're right that this is a third dead-end case beyond the parse-error one: a valid-JSON settings.json whose hooks tree has a shape `install` refuses (`mergeManagedEntries` — `"hooks"` not an object, or a Director event not an array). `ManagedEntriesPresent` returns false while `SettingsParseError` returns nil, so doctor currently says "run `director install`", which then refuses.

doctor's verdict is still correct here (FAIL) — it's the remedy text that loops, not a false-healthy. The cleanest fix reuses install's own shape validator, which means extracting it out of `mergeManagedEntries` — and this project gates changes to the settings.json merge behind explicit maintainer sign-off. So rather than duplicate the shape rules or touch the merge in this PR, it's recorded as a tracked follow-up. Deferred deliberately, not dropped.
